src/OpenSage.Game/Network/SkirmishManager.cs

```diff
@@ -50,41 +52,44 @@ public virtual void Stop()
         {
         }
 
-        private void StartGame()
+        internal void StartGame()
         {
-            var random = new Random(Settings.Seed);
-
-            var playerSettings = (from s in Settings.Slots
-                                  where s.State != SkirmishSlotState.Open && s.State != SkirmishSlotState.Closed
-                                  select new PlayerSetting(
-                                      s.StartPosition == 0 ? null : s.StartPosition,
-                                      GetItem(s.FactionIndex, _game.GetPlayableSides()),
-                                      GetItem(s.ColorIndex, _game.AssetStore.MultiplayerColors).RgbColor,
-                                      s.Team,
-                                      s.State switch
-                                      {
-                                          SkirmishSlotState.EasyArmy => PlayerOwner.EasyAi,
-                                          SkirmishSlotState.MediumArmy => PlayerOwner.MediumAi,
-                                          SkirmishSlotState.HardArmy => PlayerOwner.HardAi,
-                                          SkirmishSlotState.Human => PlayerOwner.Player,
-                                          _ => PlayerOwner.None
-                                      })).OfType<PlayerSetting?>().ToArray();
-
-            _game.StartMultiPlayerGame(
+            var playerSettings = new List<PlayerSetting>();
+            for (var i = 0; i < Settings.Slots.Length; i++)
+            {
+                var slot = Settings.Slots[i];
+
+                if (slot.State == SkirmishSlotState.Open || slot.State == SkirmishSlotState.Closed)
+                {
+                    continue;
+                }
+
+                var owner = slot.State switch
+                {
+                    SkirmishSlotState.EasyArmy => PlayerOwner.EasyAi,
+                    SkirmishSlotState.MediumArmy => PlayerOwner.MediumAi,
+                    SkirmishSlotState.HardArmy => PlayerOwner.HardAi,
+                    SkirmishSlotState.Human => PlayerOwner.Player,
+                    _ => PlayerOwner.None
+                };
+
+                playerSettings.Add(new PlayerSetting(
+                    slot.StartPosition,
+                    _game.GetPlayableSides().ElementAt(slot.FactionIndex - 1).Name,
+                    _game.AssetStore.MultiplayerColors.GetByIndex(slot.ColorIndex).RgbColor,
+                    slot.Team,
+                    owner,
+                    isLocalForMultiplayer: i == Settings.LocalSlotIndex));
+            }
+
+            _game.StartSkirmishOrMultiPlayerGame(
                 Settings.MapName,
                 Connection,
-                playerSettings,
-                Settings.LocalSlotIndex,
-                Settings.Seed);
+                playerSettings.ToArray(),
+                Settings.Seed,
+                IsNetwork);
```

Comment:
Good question - I do also use LINQ in non-critical bits of code. I think I changed this specific one just because I found it easier to set breakpoints, and perhaps it's a bit more easily readable too.